src/sdk/namespace_cache.js

```diff
@@ -486,9 +511,19 @@ class NamespaceCache {
             const hub_stream = new stream.PassThrough();
             const hub_params = { ...params, source_stream: hub_stream };
             const hub_promise = this.namespace_hub.upload_object(hub_params, object_sdk);
-
+            const hub_callback = res => {
+                if (!res.last_modified_time) {
+                    dbg.log0('Failed to set last_modified_time for cached object');
+                    return res;
+                }
+                const update_params = _.pick(_.defaults(
+                    { bucket: this.namespace_nb.target_bucket }, params), 'bucket', 'key', 'last_modified_time');
+                update_params.last_modified_time = (new Date(res.last_modified_time)).getTime();
+                setImmediate(() => object_sdk.rpc_client.object.update_object_md(update_params));
+                return res;
+            };
             // defer the final callback of the cache stream until the hub ack
-            const cache_finalizer = callback => hub_promise.then(() => callback(), err => callback(err));
+            const cache_finalizer = hub_promise.then(data => hub_callback(data), err => hub_callback(err));
```

Comment:
The type of `cache_finalizer`  should be a function and you assign it a promise - that won't work.
What we wanted to achieve was to hold the cache stream open until the hub promise completes.
So the previous code did that by setting cache_finalizer to a function that puts the cache stream final callback as the callback of the hub_promise.

Reply:
```
function stream_final_from_promise(promise) {
   return callback => promise.then(() => callback(), err => callback(err));
}

const cache_final_promise = hub_promise.then(async upload_res => {
   await object_sdk.rpc_client.object.update_object_md(update_params);
});
const cache_finalizer = stream_final_from_promise(cache_final_promise);

```